nomad/structs/csi.go

```diff
@@ -854,31 +854,38 @@ func (p *CSIPlugin) DeleteNodeForType(nodeID string, pluginType CSIPluginType) e
 	switch pluginType {
 	case CSIPluginTypeController:
 		prev, ok := p.Controllers[nodeID]
-		if ok {
-			if prev == nil {
```

Comment:
This change is failing the test because it's possible for a map of pointers to both return `ok` and a nil value. See https://play.golang.org/p/3aN8VRSkMj8. That's why we're checking for `ok` to begin with and the behavior is deliberate.

Now, arguably we should never have the nil pointer there in the first place! But there's just some unfortunate bits to our design here with both plugins and volumes. https://github.com/hashicorp/nomad/issues/8734 is an open issue to see what we can do about that.

Reply:
Ah, that makes sense! Thanks!